message_sender/views.py

```diff
@@ -84,11 +85,16 @@ class InboundViewSet(viewsets.ModelViewSet):
     """
     permission_classes = (IsAuthenticated,)
     queryset = Inbound.objects.all()
-    serializer_class = InboundSerializer
     filter_fields = ('message_id', 'in_reply_to', 'to_addr', 'from_addr',
                      'content', 'transport_name', 'transport_type',
                      'helper_metadata', 'created_at', 'updated_at',)
 
+    def get_serializer_class(self):
+        if self.action == 'create':
+            if "channel_data" in self.request.data:
+                return JunebugInboundSerializer
+        return InboundSerializer
```

Comment:
This is great, I like how simple this makes decoding different types of messages 🍰